### PR TITLE
Add markdown documentation for python-MAGNET interfacing

### LINKED ISSUE #123: Add markdown documentation of Python - MagNet critical information
PR #122 identifies two **Problems Encountered** and then describes solutions to these problems. Please write up these solutions in markdown documentation within the repository. I would like you to do this by creating a README.md file in `python/tools/magnet/`

### PR DESCRIPTION
**Pull request by Bharat-Ramadas**
# Description
This pull request captures the information provided in the `Problems Encountered` section of PR #122 description

### CODE REVIEW

File: python/tools/magnet/emach/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Desciption
+
+This is a document aimed at helping users set up their systems in order to interface python with MAGNET
+
+- In order for Python to communicate with MAGNET, we need to import the pywin32 to call ActiveX compliant programs as COM automation servers. The pywin32 module comes with Anaconda3 installation and can be imported with the command `from win32com.client import DispatchEx`
```
Comment (ngadiyar93) [inline python/tools/magnet/emach/README.md]: @Bharat-Ramadas Probably changing sentences to third person makes it more consistent with rest of the documentation in eMach?
i.e, as an example, changing `In order for Python to communicate with MAGNET, we need to import the pywin32 to call ActiveX compliant programs as COM automation servers`  to `In order for Python to communicate with MAGNET, the pywin32 needs to be imported to call ActiveX compliant programs as COM automation servers` or equivalent

You could use this as an example: https://github.com/Severson-Group/eMach/blob/develop/README.md

**Review by ngadiyar93 — CHANGES_REQUESTED** [review]
Nice work Bharat. Please see my comments.

File: python/tools/magnet/emach/README.md (hunk 1 above)
Comment (Bharat-Ramadas) [reply python/tools/magnet/emach/README.md]: @ngadiyar93 Thanks for the feedback! I'll update the markdown to more closely match the example you've shared.

Comment (Bharat-Ramadas) [reply python/tools/magnet/emach/README.md]: @ngadiyar93 Please check the updated file and let me know your thoughts on the same.

File: python/tools/magnet/emach/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Desciption
+
+This document is aimed at helping users write scripts for MAGNET in python.
+
+# Setup
+
+- In order for python to communicate with MAGNET, the pywin32 module has to be imported to call ActiveX compliant programs as Component Object Model (COM) automation servers. The pywin32 module comes with the Anaconda3 installation and can be imported with the command `from win32com.client import DispatchEx`
```
Comment (ngadiyar93) [inline python/tools/magnet/emach/README.md]: Where do I run this command- `from win32com.client import DispatchEx`?
Should the user run this in Anaconda Prompt? Or should this be added somewhere in the script? Or does it work either way?
May be a good idea to make it explicit?.

File: python/tools/magnet/emach/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- If the importing of the module raises an error flag, please make sure that the path of the Anaconda3 site-packages has been added to the system's `Environment variables` as `PYTHONPATH`.
```
Comment (ngadiyar93) [inline python/tools/magnet/emach/README.md]: May be better to say `If an error occurs and the module import is unsuccessful, please make sure.......` ?

File: python/tools/magnet/emach/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+   MN.Visible = True #Makes MAGNET window visible
+   Doc = MN.newDocument()
+   View = Doc.getView()
+   Sol = Doc.getSolution()
+   MNConsts = MN.getConstants() 
+```
+# Function Returns
+
+MAGNET APIs can be divided into 2 categories based on how the function's return value can be accessed:
+- Functions who's return values can be directly accessed by assigining the function call to a variable fall under the first category. `makeSimpleCoil` is an example of such a function. Its return value can be obtained in python directly with the following syntax: `coil = Doc.makeSimpleCoil(ProblemID, ArrayOfValues)`
```
Comment (ngadiyar93) [inline python/tools/magnet/emach/README.md]: I guess you mean to say "whose" and not "who's" ?

File: python/tools/magnet/emach/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    MN.processCommand("REDIM strArray(0)")
+    MN.processCommand("DIM pType")
+    MN.processCommand("pType = getDocument.getParameter(\"{}\", \"{}\", strArray)".format(path,parameter))
+    MN.processCommand('Call setVariant(0, strArray,"PYTHON")')    
+    param = MN.getVariant(0,"PYTHON")
+    MN.processCommand('Call setVariant(0, pType,"PYTHON")')    
+    param_type = MN.getVariant(0,"PYTHON")
+    return param, param_type
+```
+The function definition of MAGNET APIs can be obtained from MAGNET's Help Topics.
```
Comment (ngadiyar93) [inline python/tools/magnet/emach/README.md]: @Bharat-Ramadas Do you think the MagNet knowledgebase article with the Python scripting example would be useful? If so, could you also add a link to it?

**Review by ngadiyar93 — CHANGES_REQUESTED** [review]
Nice Work Bharat!. Please see my comments.

**Review by elsevers — CHANGES_REQUESTED** [review]
- [x] Clean up grammar and some capitalization (`Python` and `MagNet`)
- [x] Provide more narrative framing to make this an easier read at the start of each section
- [x] Revise purpose statement to clarify that this doc is intended for developers and make the language more direct.

**Comment by elsevers** [comment]
Why is this in an `emach` subfolder? We don't want to have a folder in the repo called `emach`

**Review by elsevers — APPROVED** [review]

**Review by ngadiyar93 — APPROVED** [review]
Looks good to me!

**Pull request merged**